### PR TITLE
Stop the alarm on open notification

### PR DESCRIPTION
**Pull request by LesYampolskyi**
The purpose of this pull request is to stop the alarm's ringing/vibration when the user opens the local notification.

### CODE REVIEW

**Comment by gdelataillade** [comment]
Thank you @LesYampolskyi !

Do you think this has to be the default behavior ? Shouldn't we add a parameter to disable it ? What do you think ?

File: lib/alarm.dart [hunk 1]
```diff
@@ -9,6 +9,7 @@ import 'package:alarm/src/android_alarm.dart';
 import 'package:alarm/service/notification.dart';
 import 'package:alarm/service/storage.dart';
 import 'package:flutter/foundation.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
```
Comment (gdelataillade) [inline lib/alarm.dart:12]: I would prefer if the `flutter_local_notifications` code is only called from the `AlarmNotification` service. Would that be possible ?

Comment (LesYampolskyi) [reply lib/alarm.dart]: Yes, it is totally possible.

I imported flutter_local_notifications package to lib/alarm only for the NotificationResponse type in onSelectNotification. Any code from flutter_local_notifications is not used.

I thought about circular imports. lib/alarm uses lib/services/notification and that's why i placed onSelectNotification in lib/alarm.
But i see that lib/services/notification also uses lib/alarm (alarmPrint) and [the circular dependencies are allowed for dart](https://stackoverflow.com/questions/65324492/dart-circular-imports)

I replaced the callback and added a new commit.

**Comment by LesYampolskyi** [comment]
> 

Maybe it is better to make it changeable and add a parameter to disable it. By default, it is true, but the developer can change it. For example, if the business logic requires some extra steps for the user after opening the app and doing something manually.
On Android and iOS alarms, there is also a snooze option. Perhaps this one should be added as well.

File: lib/service/notification.dart [hunk 2]
```diff
@@ -17,22 +17,42 @@ class AlarmNotification {
...
+    final initializationSettingsIOS = DarwinInitializationSettings(
       requestAlertPermission: false,
       requestBadgePermission: false,
       requestSoundPermission: false,
+      onDidReceiveLocalNotification: (
+        int? id,
+        String? title,
+        String? body,
+        String? payload,
+      ) async {},
```
Comment (gdelataillade) [inline lib/service/notification.dart:32]: This callback is empty. Is it possible to remove it so we can make `initializationSettingsIOS` a const again ?

File: lib/alarm.dart (hunk 1 above)
Comment (gdelataillade) [reply lib/alarm.dart]: Awesome thank you.

File: lib/service/notification.dart (hunk 2 above)
Comment (LesYampolskyi) [reply lib/service/notification.dart]: This callback is only applicable to iOS versions older than 10. 
I've added a <code>onSelectNotificationOldIOS</code> to this callback. It uses a different set of arguments so I can't reuse <code>onSelectNotification</code>

**Comment by LesYampolskyi** [comment]
@gdelataillade how should the parameter be named? Is <code>stopAlarmOnReceivedNotification</code> ok?

And I noticed in <code>alarm_settings.dart</code> that <code>enableNotificationOnKill</code> is missing in <code>copyWith</code>, <code>toJson</code>  and <code>factory AlarmSettings.fromJson</code>. As far as I understand they used to recreate the alarms after app is killed and should this parameter be added as well?

**Comment by gdelataillade** [comment]
Yes `enableNotificationOnKill` is missing, I forgot to add it thanks !

I suggest to call the parameter `stopOnNotificationOpen`. What do you think ?

**Comment by LesYampolskyi** [comment]
> Yes `enableNotificationOnKill` is missing, I forgot to add it thanks !
> 
> I suggest to call the parameter `stopOnNotificationOpen`. What do you think ?

Sounds good.

**Comment by LesYampolskyi** [comment]
@gdelataillade I added the parameter <code>stopOnNotificationOpen</code> for <code>AlarmSettings</code> and modified <code>README.md</code>. And also added the missed <code>enableNotificationOnKill</code>.

I

**Review by gdelataillade — APPROVED** [review]

**Comment by gdelataillade** [comment]
@LesYampolskyi sounds good

Have you tested on Android ?

**Comment by LesYampolskyi** [comment]
> @LesYampolskyi looks good
> 
> Have you tested on Android ?

Yes, it works

**Comment by gdelataillade** [comment]
Available in `1.0.3` update.

**Pull request merged**